### DISCUSSION TITLE
Proton - tweaks (real compact mode)

### DISCUSSION THREAD

**Discussion by Aris-t2** [Aris-t2/CustomCSSforFx#458] [General]
Has anyone already tested "Proton" UI? Mozilla planned to remove "compact mode", but backed down.

After looking into what they call "compact" in Proton UI, I decided to run some tweaks and tests. The result was a "more compact mode". The code relies on latest Nightly builds and some `about:config` preferences:
- toolkit.legacyUserProfileCustomizations.stylesheets > true
- browser.proton.appmenu.enabled > true
- browser.proton.enabled > true
- browser.proton.tabs.enabled > true
- browser.uidensity > 1

[real_compact_mode.css](https://github.com/Aris-t2/CustomCSSforFx/blob/master/proton/real_compact_mode.css)

Besides code tweaks to change toolbar, tabs and app menu the "real compact mode" additionally uses CSS code from existing CustomCSSforFx settings: hiding megabar breakout, compact results and very classic results appearance (title+url).

![Compact](https://user-images.githubusercontent.com/6013569/111913679-860e0200-8a6f-11eb-94ef-3ec0fb3a4ff6.jpg)

**Comment by Acid-Crash** [comment]
Hi @Aris-t2 
I did check the Proton UI.
As for now I am using Compact density on release, so if they will not remove Compact in Proton I can live with it.
The one point that bugs me most about Proton is the context menu in App button, they are way to big.

I have also checked real_compact_mode.css
Well as the name states it is REAL compact.
Way to compact for me, but of course this is a matter of personal taste

**Comment by marty60** [comment]
How I wish mozilla would just leave the uI alone and concentrate on more important things but it's not in their nature sadly.

For now I'm only seeing two issues: the bookmark toolbar on the bottom no longer autohides and the custom button icons (from the legacy custom button addon) are about as big as a breadcrumb while webextension icons remain the same. I don't mind them small but they could stand to be made a little bigger. 

For now, proton is disabled but at some point when the current layout is gone I'm sure you will have lots of requests from users like me wanting to get back to normal.

**Comment by Aris-t2** [comment]
At some point I will release CSS code to undo (some of) the Proton UI changes.

**Comment by nicolaasjan** [comment]
> Wouldn't it be better for inactive tabs to have outlines and/or different colour to make them better to distinguish, particularly when there are many tabs open?

Never mind, I forgot to set `toolkit.legacyUserProfileCustomizations.stylesheets` to `true` (new profile). Haha.
There is an actual outline.

Sorry for posting this in the wrong place.

**Comment by Aris-t2** [comment]
Of course it would be better, but this is not about improving Proton UI.
It is about offering a "more compact" experiment.

**Comment by nicolaasjan** [comment]
> Of course it would be better, but this is not about improving Proton UI.
> It is about offering a "more compact" experiment.

You're right of course.
I edited my comment.

Thanks for your hard work!

**Comment by Hlsgs** [comment]
@Aris-t2 Have you tried to get multi-row tabs working with Proton and these "compact mode" tweaks?

**Comment by Aris-t2** [comment]
No I have not tested multi row tabs with protons compact mode.
It was just a tiny experiment without being connected to this projects main code.

**Comment by Hlsgs** [comment]
> 
> 
> No I have not tested multi row tabs with protons compact mode.
> It was just a tiny experiment without being connected to this projects main code.

How about multi-row tabs with vanilla, non-compact? Any intent tu update that part of the project?

**Comment by Aris-t2** [comment]
Multiple tab lines should work using this code: https://github.com/Aris-t2/CustomCSSforFx/blob/master/proton/multiple_tab_lines.css

(It is a minor modification of https://github.com/Aris-t2/CustomCSSforFx/blob/master/classic/css/tabs/tabs_multiple_lines_fx81.css)

**Comment by Hlsgs** [comment]
> 
> 
> Multiple tab lines should work using this code: https://github.com/Aris-t2/CustomCSSforFx/blob/master/proton/multiple_tab_lines.css
> 
> (It is a minor modification of https://github.com/Aris-t2/CustomCSSforFx/blob/master/classic/css/tabs/tabs_multiple_lines_fx81.css)

Awesome, thanks! Two caveats, both regarding muted tabs:
- it should hide the "MUTED" text in order to work properly with non-compact modes
- in compact mode, however long that will last, it should move the muted icon from above the favicon(have to congratulate Mozilla on this one, pure genius :) )

**Comment by Aris-t2** [comment]
Not sure I can follow. Using the code, while not having current 'compact mode' active, is not intended, at least for now. 

![image](https://user-images.githubusercontent.com/6013569/119219246-dee82e00-bae4-11eb-816a-e2b9ec523489.png)

**Comment by Hlsgs** [comment]
> 
> 
> Not sure I can follow. Using the code, while not having current 'compact mode' active, is not intended, at least for now.
> 
> ![image](https://user-images.githubusercontent.com/6013569/119219246-dee82e00-bae4-11eb-816a-e2b9ec523489.png)

As you can on the muted tab in that picture, the "muted" icon is covering the favicon of the site, which is bad UX. I personally keep all tabs muted by default and only unmute the ones i need, so this would be a complete no-go for me. Since Mozilla do not officially support "compact mode" anymore, I doubt they'll ever fix that, so can it be moved to the left of the close button, where it used to be?

Unrelated: I think https://github.com/Aris-t2/CustomCSSforFx/blob/b368cf808bbb22f391b9871ff1455e39245b4cfb/proton/multiple_tab_lines.css#L53 should account for `--proton-tab-block-margin` in order to be able to show the set `--tabs-lines` number of lines without showing the scrollbar.

**Comment by Aris-t2** [comment]
You can not move the icon out of the favicon box.

This should hide the icon in general:

```css
.tab-icon-overlay{
  display: none !important;
}

```
And this only when muted.

```css
.tab-icon-overlay[muted="true"]{
  display: none !important;
}
```

**Comment by AdamTT** [comment]
Hi Aris,
I use your .css file:
[real_compact_mode.css](https://github.com/Aris-t2/CustomCSSforFx/blob/master/proton/real_compact_mode.css)
and get great result. 

![1](https://user-images.githubusercontent.com/18615118/120538569-f7aaea80-c3e6-11eb-81c1-2ea490274213.jpg)

But there is one inconvienience. 
When I add another tab (or change window size to smaller one) the 'close-tab_button' (X) disapear on all non-active tabs.
Is there a way to change the minimum tab width (or something else) to avoid this?

![2](https://user-images.githubusercontent.com/18615118/120540454-02667f00-c3e9-11eb-83f9-4f4c08bfc756.jpg)

TIA,
Adam

**Comment by Aris-t2** [comment]
This is default browser behavior.

You can override it using this code:

```css
/* always show 'close tab icon' */
.tabbrowser-tab:not([pinned="true"]) > .tab-stack > .tab-content > .tab-close-button:not([selected="true"]) {
  display: initial !important;
}
```

**Comment by AdamTT** [comment]
Thanks a lot. :)
Modification works like a charm.

Maybe I can try to ask for another little thing? Is it possible to distinguish somehow not_activ/not_hovered tabs each to other? Outline? Vertical separator?

**Comment by Aris-t2** [comment]
```css
.tabbrowser-tab:not(:hover) > .tab-stack > .tab-background:not([selected=true], [multiselected]) {
  background-color: color-mix(in srgb, currentColor 11%, transparent);
}

.tabbrowser-tab:hover > .tab-stack > .tab-background:not([selected=true], [multiselected]) {
  background-color: color-mix(in srgb, currentColor 17%, transparent) !important;
}
```

**Comment by AdamTT** [comment]
Thanks again, Aris!
Now my FF looks awsome for me.

**Comment by speefak** [comment]
The Proton real-compact-mode is really compact ;). But customization is not able any more. So i created a simple proton compact mode focused on screen height and width in relation of tabtoolbar, tabs down below and simple borderless buttons for those who works with a lot of pinned tab and open tabs. 

Tested on Debian 10/11 | FF 91.0.1 | should work on all linux distributions - OSX and Windows not tested

<a href='https://postimg.cc/qzYXcMTt' target='_blank'><img src='https://i.postimg.cc/qzYXcMTt/Bildschirmfoto-von-2021-08-22-15-59-42.png' border='0' alt='Bildschirmfoto-von-2021-08-22-15-59-42'/></a>

HowTo : https://speefak.spdns.de/oss_lifestyle/firefox-style-framework/ (german)
Detailed HowTo  : https://github.com/Aris-t2/CustomCSSforFx/issues/406#issuecomment-903268401 (english)
OSX folder icon file : https://i.postimg.cc/Ng9FxbpQ/folder-item.png?dl=1 ( replace ./image/folder-item.png )
Preconfigured framework : https://speefak.spdns.de/oss_lifestyle/wp-content/uploads/2021/08/2021-08-22_chrome.7z
Detailed HowTo : https://speefak.spdns.de/oss_lifestyle/firefox-style-framework/

TODO :
- remove hidden hamburger button empty space 
- set old style tab button style ( ESR style )
- reduce tab button height.

aris: i suggest to create a new css file : ./css/tabs/tabs_below_navigation_toolbar_compact.css and linked from userChrome line 326 => @import "./css/tabs/tabs_below_navigation_toolbar_compact.css"; /**/

**Comment by Pizzapops** [comment]
@Aris-t2 
The real compact mode is working fine for me. I have been using compact bookmark & history panels for years.
![2021-09-02_171905](https://user-images.githubusercontent.com/33039961/132045478-182df1a4-80aa-4376-8977-0e0b81cc67f1.jpg)
I tweaked popup_compact_menus.css to a better match to the  app menu.
![2021-09-02_171901](https://user-images.githubusercontent.com/33039961/132046858-c0dce550-f2c3-443b-89c6-34008b794ded.jpg)
```css
/* compact lines in bookmarks & history */
.sidebar-placesTree treechildren::-moz-tree-row {
    min-height: unset !important;
    height: 1.5em !important;
    border-width: 1px !important;
}

/* popup_compact_menus.css lines 19-25*/
menupopup > menuitem,
menupopup > menu {
  padding-block: 0 !important;
  padding-inline-start: 4px !important;
    min-height: unset !important;
    height: 1.5em !important;
}

/*** Set custom colors for popup menu color variables ***/
/* thanks to jscher2000 & black7375*/
menupopup {
  --menu-color: #000 !important;
  --menu-disabled-color: color-mix(in srgb, var(--menu-color) 50%, transparent) !important;
  --menu-background-color: #fff !important;
  --menuitem-hover-background-color: #d4d4d4 !important;
}

Since ver 91 I have had problems with popup menus being black in themes where they previously they were white. 
I tried using different color variables without luck. Did I miss something? I finally used some css to fix it.

**Comment by Speravir** [comment]
I have `generalui/popup_compact_menus.css` activated, but this had no effect on the Bookmarks toolbar menu in Fx92. In addition to the variable changes posted by Aris I’ve added and adjusted the following rule (came from another source, this may be too specific):
```css
#PlacesToolbar .bookmark-item :is(menu, menuitem) {
	margin: 0 2px 0 4px !important;/* 2px left, 4px right */
	min-height: 20px !important;
}
```

I already had added this style which overrules some code from `generalui/popup_compact_menus.css` (values from there in comment):
```css
menupopup > :is(menu, menuitem) {
	padding-block: 1px !important;/* 2px */
	padding-inline-start: 0 !important;/* 4px */
}/* has an influence on symbol position */
```

**Comment by Speravir** [comment]
Too fast with sending …

@Pizzapops could you try to fix your message above? The sentences
> Since ver 91 I have had problems with popup menus being black in themes where they previously they were white. 
I tried using different color variables without luck. Did I miss something? I finally used some css to fix it.

Are in the code block, but I am sure this was not your intention. (Forgot twice a backtick?)

**Comment by Aris-t2** [comment]
I'm not sure as I did not face this issue yet. Does this also happen without any other code?

I just posted a new file to fix the issues with Fx97+/ESR 91.6.

**Comment by mycek91** [comment]
Hi @Aris-t2 
There is a problem with the latest update Firefox 126 and the real_compact_mode.css.
I know this mode is considered legacy, but can you have a look at it.
It is so comfortable.
There is a bug that causes the address bar to overlap (hide) with suggestion window when I try to search for something in it.
There is a screenshot
![image](https://github.com/Aris-t2/CustomCSSforFx/assets/40513214/f0acb419-a9ba-4338-9509-5c5c9e8b5c9e)

**Reply by Aris-t2** [reply]
There were some issue with recent updates regarding the location bar.

Ty to replace the code for urlbar with this one (it was fixed for Fx 126+):

https://github.com/Aris-t2/CustomCSSforFx/blob/master/current/css/locationbar/megabar_expanding_breakout_disabled.css

**Reply by mycek91** [reply]
Everything works again!
Many Thanks
